components/fxa-client/ffi/src/lib.rs

```diff
@@ -289,6 +289,17 @@ pub extern "C" fn fxa_get_access_token(
     })
 }
 
+/// This method should be called when a request made with
+/// an OAuth token failed with an authentication error.
+/// Currently it only clears the OAuth tokens cache, so the
+/// caller can try to call `fxa_get_access_token` or `fxa_profile`
+/// again.
+#[no_mangle]
```

Comment:
From conversation with @linuxwolf, it may be worth trying to clarify the expectations on the caller a little more here. It doesn't have to be called immediately, as long as you call it before trying to get another access token. So for example, it's same to call this at the end of a sync, rather than having to arrange for it to be called *immediately* when the error occurs.